gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final DnsRecord AAAA_RECORD_ZONE1 = DnsRecord
+      .builder("www." + ZONE1.dnsName(), DnsRecord.Type.AAAA)
+      .records(ImmutableList.of("ed:ed:12:aa:36:3:3:105"))
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final ChangeRequest CHANGE_ADD_ZONE1 = ChangeRequest.builder()
+      .add(A_RECORD_ZONE1)
+      .add(AAAA_RECORD_ZONE1)
+      .build();
+  public static final ChangeRequest CHANGE_DELETE_ZONE1 = ChangeRequest.builder()
+      .delete(A_RECORD_ZONE1)
+      .delete(AAAA_RECORD_ZONE1)
+      .build();
+
+  public static void purge() {
+    Page<Zone> zones = DNS.listZones();
+    Iterator<Zone> zoneIterator = zones.iterateAll();
+    while (zoneIterator.hasNext()) {
+      Zone zone = zoneIterator.next();
+      List<DnsRecord> toDelete = new LinkedList<>();
+      if (zone.name().startsWith(PREFIX)) {
+        Iterator<DnsRecord> dnsRecordIterator = zone.listDnsRecords().iterateAll();
+        while (dnsRecordIterator.hasNext()) {
+          DnsRecord record = dnsRecordIterator.next();
+          if (!ImmutableList.of(DnsRecord.Type.NS, DnsRecord.Type.SOA).contains(record.type())) {
+            toDelete.add(record);
+          }
+        }
+        if (!toDelete.isEmpty()) {
+          zone.applyChangeRequest(ChangeRequest.builder().deletions(toDelete).build());
+        }
+        zone.delete();
+      }
+    }
+  }
+
+  private static List<Zone> filter(Iterator<Zone> iterator) {
+    List<Zone> result = new LinkedList<>();
+    while (iterator.hasNext()) {
+      Zone zone = iterator.next();
+      if (zone.name().startsWith(PREFIX)) {
+        result.add(zone);
+      }
+    }
+    return result;
+  }
+
+  @BeforeClass
+  public static void before() {
+    purge();
+  }
+
+  @AfterClass
+  public static void after() {
+    purge();
+  }
+
+  @Test
+  public void testCreateValidZone() {
+    Zone created = DNS.create(ZONE1);
+    assertEquals(ZONE1.description(), created.description());
+    assertEquals(ZONE1.dnsName(), created.dnsName());
+    assertEquals(ZONE1.name(), created.name());
+    assertNotNull(created.creationTimeMillis());
+    assertNotNull(created.nameServers());
+    assertNull(created.nameServerSet());
+    assertNotNull(created.id());
+    Zone retrieved = DNS.getZone(ZONE1.name());
+    assertEquals(created, retrieved);
+    created = DNS.create(ZONE_EMPTY_DESCRIPTION);
+    assertEquals(ZONE_EMPTY_DESCRIPTION.description(), created.description());
+    assertEquals(ZONE_EMPTY_DESCRIPTION.dnsName(), created.dnsName());
+    assertEquals(ZONE_EMPTY_DESCRIPTION.name(), created.name());
+    assertNotNull(created.creationTimeMillis());
+    assertNotNull(created.nameServers());
+    assertNull(created.nameServerSet());
+    assertNotNull(created.id());
+    retrieved = DNS.getZone(ZONE_EMPTY_DESCRIPTION.name());
+    assertEquals(created, retrieved);
+  }
+
+  @After
+  public void tearDown() {
+    purge();
+  }
```

Comment:
Provided that the user's project does not contain millions of other zones, it saves work only for tests where no records are created. For zones where records are created, selective deletion sounds like a bad trade of readable code for a bit of performance (you cannot delete zone if it contains some records, so you would need to check what exists anyways). For tests where no dns records are created, we can do this. 

Thus, I will do selective deletion where no dns record are created and I will call `clean()` in tests where we do so.

I will also add `@BeforeClass` with clean.